evmjit/libevmjit/BasicBlock.cpp

```diff
@@ -43,6 +44,7 @@ BasicBlock::LocalStack::LocalStack(BasicBlock& _owner) :
 
 void BasicBlock::LocalStack::push(llvm::Value* _value)
 {
+	assert(_value->getType() == Type::Word);
```

Comment:
I dont know how save this assert is. But to avoid hard crashes of the client, you could use asserts as provided by Gav in libdevcore/Common.h to smoothly fail.